docs/proposals/resource-aggregation-proxy/README.md

```diff
@@ -0,0 +1,455 @@
+---
+title: Resource Aggregation Proxy
+authors:
+- "@ikaven1024"
+- "@cmicat"
+reviewers:
+- TBD
+approvers:
+- TBD
+
+creation-date: 2022-6-24
+
+---
+
+# Resource Aggregation Proxy
+
+# Summary
+
+![without-karmada](without-karmada.svg)
+
+Before using Karmada, there is only one kubernetes cluster per IDC. The deployment system, monitor & alarm system, service discovery and various other systems access raw kubernetes resources directly, including workload, configmap, secret, pod, node.
+
+![with-karmada](with-karmada.svg)
+
+We can't access pod and node information from Karmada API server. So after introducing Karmada, the PaaS components can no longer access pod and node resources. This will break the PaaS components. 
+
+
+
+In order to solve the problem, we can either:
+
+1. **Modify all of PaaS components.** Force all of them understand Karmada's logic, and find the pod node in the appropriate cluster by themselves.
+2. **Provide a Resource Aggregation Proxy.** All PaaS components just ask the proxy to give them the pod node information. The proxy will find the right clusters to get data, merge data, then return it to the client.
+
+
+
+Apparently, using the proxy will greatly simplify external components' access to kubernetes clusters, and hide multiple cluster's implementation details from them. That's what we propose.
+
+
+
+
+Try these:
+
+- **[Aggregated Kubernetes API Endpoint](https://github.com/karmada-io/karmada/blob/master/docs/userguide/aggregated-api-endpoint.md)** This component is built for Unified authentication. The restful api client still needs to enumerate all the member clusters to get the right information.
+- **[Caching](https://github.com/karmada-io/karmada/tree/master/docs/proposals/caching) (aka karmada-search)** This component can only get/list, not watch or update data. 
+
+# Motivation
+
+Write a Resource Aggregation Proxy for Karmada. This proxy provides native kubernetes API for components outside kubernetes. When the client requests operations (including reads & writes), the proxy will figure out the correct downstream kubernetes API servers (including Karmada API server and member clusters), request all the selected servers, and merge the responses into one before responding to the client. External components can treat Karmada-managed kubernetes clusters as one single cluster.
+
+## Goals
+
+Allow kubernetes API clients (kubectl, client-go, and other clients that use kubernetes APIs) to read/write the following resources:
+
+- workload: create/get/list/watch/update/patch/delete
+- node: get/list/watch/update/patch
+- pod: get/list/watch/update/patch/logs/exec
+
+## Non-Goals
+
+- read/write resources:
+  - service account token
+  - event
+
+- Provide a solution for unique resource names across all member clusters.
```

Comment:
Yeah, I get it from below section:

> Resource names need to be unique across all member clusters. Otherwise, the proxy will not be able to differentiate which resource needs to be accessed.

But how to assured that the resource name is unique?

Reply:
It is out the scope of this proposal. But i have a rough idea.
We can add a webhook in member clusters, used to validate the resource creations. When validating creations, webhook will search the name from `proxy` component in controller panel.

Reply:
> We can add a webhook in member clusters, used to validate the resource creations. When validating creations, webhook will search the name from proxy component in controller panel.

Personally, I don't think people would like it. a little bit complicated I guess.